dali/pipeline/operator/common.h

```diff
@@ -137,19 +132,16 @@ void GetGeneralizedArg(span<T> result, const std::string name, int sample_idx, c
 }
 
 template <typename ArgumentType, typename ExtentType>
-std::pair<int, int> GetShapeLikeArgument(std::vector<ExtentType> &out_shape,
-                                         const OpSpec &spec,
+std::pair<int, int> GetShapeLikeArgument(std::vector<ExtentType> &out_shape, const OpSpec &spec,
                                          const std::string &argument_name,
-                                         const ArgumentWorkspace &ws,
-                                         int ndim = -1,
-                                         int batch_size = -1 /* -1 = get from "batch_size" arg */) {
-  if (batch_size < 0)
-    batch_size = spec.GetArgument<int>("batch_size");
+                                         const ArgumentWorkspace &ws, int ndim = -1,
+                                         int batch_size = -1) {
```

Comment:
Weird grouping of arguments. Either keep them one per line or use logical grouping, like:
out_shape  // output
spec, arg_name, ws,  // source of the argument
ndim, batch_size  // shape